### PR TITLE
Windows gmsa e2e: Don't assume bash is avaliable for webhook deployment

### LINKED ISSUE #108816: Windows GMSA Full e2e test uses bash to install the plugin
### What happened?

Cluster api provides a framework to run the e2e tests which uses the conformance image:

https://github.com/kubernetes-sigs/cluster-api/blob/692497dbb340beecc04171ccd73f370f173da3a9/test/framework/kubetest/run.go#L43-L44

The gmsa tests use bash to install the webhook:

https://github.com/kubernetes/kubernetes/blob/4d08582d1fa21e1f5887e73380001ac827371553/test/e2e/windows/gmsa_full.go#L339-L344

The test will fail when run in the conformance image because of the bash:

```
• Failure [9.850 seconds]                                                                                         
[sig-windows] [Feature:Windows] GMSA Full [Serial] [Slow]                                                         
test/e2e/windows/framework.go:27                                                                                  
  GMSA support                                                                                                    
  test/e2e/windows/gmsa_full.go:96                                                                                
    can read and write file to remote SMB folder [It]                                                             
    test/e2e/windows/gmsa_full.go:174                                                                             

    Mar 18 22:43:47.188: unable to deploy GMSA webhook, output:                                                   
    : exec: "bash": executable file not found in $PATH     
```

### What did you expect to happen?

we should install the component another way

### How can we reproduce it (as minimally and precisely as possible)?

run this e2e conformance image with `--ginkgo.focus="GMSA.Full"`

### Anything else we need to know?

/sig windows
fyi @iXinqi

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by jsturtevant**
#### What type of PR is this?
/kind failing-test

#### What this PR does / why we need it:

Assuming bash is installed when running the e2e tests fails when running inside the conformance image. This provides a way to deploy the webhook that isn't dependent on the tools installed in the local machine running `e2e.test`.  

#### Which issue(s) this PR fixes:

Fixes #108816

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig windows
/assign @marosset @knabben @ixinqi @claudiubelu

### CODE REVIEW

**Comment by jsturtevant** [comment]
this would require: https://github.com/kubernetes-sigs/windows-gmsa/pull/68

File: test/e2e/windows/gmsa_full.go [hunk 1]
```diff
@@ -78,7 +81,7 @@ const (
 
 	// gmsaWebhookDeployScriptURL is the URL of the deploy script for the GMSA webook
 	// TODO(wk8): we should pin versions.
-	gmsaWebhookDeployScriptURL = "https://raw.githubusercontent.com/kubernetes-sigs/windows-gmsa/master/admission-webhook/deploy/deploy-gmsa-webhook.sh"
+	gmsaWebhookDeployScriptURL = "https://raw.githubusercontent.com/jsturtevant/windows-gmsa/b2e7486615c7fab5a6937c80d037f8d7f2909469/admission-webhook/deploy/deploy-gmsa-webhook.sh"
```
Comment (knabben) [inline test/e2e/windows/gmsa_full.go:84]: https://github.com/kubernetes-sigs/windows-gmsa/pull/68

File: test/e2e/windows/gmsa_full.go [hunk 2]
```diff
@@ -419,7 +457,7 @@ func createRBACRoleForGmsa(f *framework.Framework) (string, func(), error) {
 
 // createServiceAccount creates a service account, and returns its name.
 func createServiceAccount(f *framework.Framework) string {
-	accountName := f.Namespace.Name + "-sa"
+	accountName := f.Namespace.Name + "-sa-" + string(uuid.NewUUID())
```
Comment (knabben) [inline test/e2e/windows/gmsa_full.go:460]: Do we need to randomize the SA account name?

Comment (jsturtevant) [reply test/e2e/windows/gmsa_full.go:460]: Not necessarily, I was re-using this function which sets up the service account for the gMSA pod for the deployment which needed additional permissions on the API server.  This allowed me to re-use it.

**Comment by jsturtevant** [comment]
This worked when running inside the conformance image built from this branch:

```
docker run --rm -it -v $PWD/kubeconfig.windows.ctrd:/tmp/kubeconfig --entrypoint /usr/local/bin/e2e.test jsturtevant/conformance:gmsa --provider=skeleton --node-os-distro=windows --ginkgo.focus="GMSA.Full" -ginkgo.skip="\\[LinuxOnly\\]|\\[Excluded:WindowsDocker\\]|device.plugin.for.Windows" --kubeconfig=/tmp/kubeconfig
```

```
Mar 23 22:38:05.992: INFO: Running '/usr/local/bin/kubectl --kubeconfig=/tmp/kubeconfig --namespace=gmsa-full-test-windows-7873-webhook delete pod webhook-deployer.gmsa-full-test-windows-7873'
Mar 23 22:38:06.130: INFO: rc: 1
[AfterEach] [sig-windows] [Feature:Windows] GMSA Full [Serial] [Slow]
  test/e2e/framework/framework.go:186
Mar 23 22:38:06.130: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
STEP: Destroying namespace "gmsa-full-test-windows-7873" for this suite.

• [SLOW TEST:31.710 seconds]
[sig-windows] [Feature:Windows] GMSA Full [Serial] [Slow]
test/e2e/windows/framework.go:27
  GMSA support
  test/e2e/windows/gmsa_full.go:99
    can read and write file to remote SMB folder
    test/e2e/windows/gmsa_full.go:170
------------------------------
{"msg":"PASSED [sig-windows] [Feature:Windows] GMSA Full [Serial] [Slow] GMSA support can read and write file to remote SMB folder","total":2,"completed":2,"skipped":6762,"failed":0}
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSMar 23 22:38:06.287: INFO: Running AfterSuite actions on all nodes
Mar 23 22:38:06.287: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage.glob..func18.2
Mar 23 22:38:06.287: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage.glob..func8.2
Mar 23 22:38:06.287: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage.glob..func7.2
Mar 23 22:38:06.287: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage/vsphere.glob..func17.3
Mar 23 22:38:06.287: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage/vsphere.glob..func9.2
Mar 23 22:38:06.287: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage/vsphere.glob..func4.2
Mar 23 22:38:06.287: INFO: Running Cleanup Action: k8s.io/kubernetes/test/e2e/storage/vsphere.glob..func1.3
Mar 23 22:38:06.287: INFO: Running AfterSuite actions on node 1
{"msg":"Test Suite completed","total":2,"completed":2,"skipped":7124,"failed":0}

Ran 2 of 7126 Specs in 64.817 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 7124 Skipped
PASS
```

**Review by knabben — APPROVED** [review]

File: test/e2e/windows/gmsa_full.go [hunk 3]
```diff
@@ -77,7 +76,6 @@ const (
 	gmsaCustomResourceName = "gmsa-e2e"
 
 	// gmsaWebhookDeployScriptURL is the URL of the deploy script for the GMSA webook
-	// TODO(wk8): we should pin versions.
 	gmsaWebhookDeployScriptURL = "https://raw.githubusercontent.com/kubernetes-sigs/windows-gmsa/master/admission-webhook/deploy/deploy-gmsa-webhook.sh"
```
Comment (marosset) [inline test/e2e/windows/gmsa_full.go:79]: did this get updated to use the gcr.io images we started building?

Comment (marosset) [reply test/e2e/windows/gmsa_full.go:79]: nope :(
https://github.com/kubernetes-sigs/windows-gmsa/blob/0c81292f090d2aa7abd24cd0f5a62de02ba4a52e/admission-webhook/deploy/deploy-gmsa-webhook.sh#L25

Let me make an issue for that in the gmsa repo

Comment (marosset) [reply test/e2e/windows/gmsa_full.go:79]: https://github.com/kubernetes-sigs/windows-gmsa/issues/70

Comment (marosset) [reply test/e2e/windows/gmsa_full.go:79]: yea, that shouldn't block this PR,

**Pull request merged**